tasks/install_shell.sh

```diff
@@ -35,6 +35,19 @@ exists() {
   fi
 }
 
+# Check whether the apt config file has been modified, warning and exiting early if it has
+assert_unmodified_apt_config() {
+  dpkg-query -W -f='${Conffiles}\n' '*' | awk 'OFS="  "{print $2,$1}' | md5sum -c 2>/dev/null | awk -F': ' '$2 !~ /OK/{print $1}' | grep -q puppet.list
+
+  local result=$?
+  local file="/etc/apt/sources.list.d/puppet.list"
+
+  if [ -f "$file" ] && [ "$result" -eq 0 ]; then
```

Comment:
The file check is here since the previous command will report that the file was modified if it doesn't exist. If the file doesn't exist, then there won't be a conflict with a manually maintained file, so we should go ahead and install the repo and agent.